src/main/java/org/folio/rest/impl/InvoiceLineHelper.java

```diff
@@ -88,17 +89,79 @@
     return future;
   }
 
+  private void updateOutOfSyncInvoiceLine(InvoiceLine invoiceLine) {
+    VertxCompletableFuture.runAsync(ctx, () -> {
+      InvoiceLineHelper helper = new InvoiceLineHelper(okapiHeaders, ctx, lang);
+      helper.updateInvoiceLineToStorage(invoiceLine)
+        .handle((ok, fail) -> {
+          helper.closeHttpClient();
+          return null;
+        });
+    });
+  }
+
+  /**
+   * Calculate invoice line total and compare with original value if it has changed
+   *
+   * @param invoiceLineFromStorage invoice line to update totals for
+   * @return {code true} if grand total value is different to original one
+   */
+  public CompletableFuture<Boolean> reCalculateInvoiceLineTotals(InvoiceLine invoiceLineFromStorage) {
+
+    // 1. Get original values
+    Double existingTotal = invoiceLineFromStorage.getTotal();
+
+    // 2. Recalculate totals
+    return calculateInvoiceLineTotals(invoiceLineFromStorage).thenApply(invoiceLineWithTotalRecalculated -> {
+      Double recalculatedTotal = invoiceLineWithTotalRecalculated.getTotal();
+
+      // 3. Compare if anything has changed
+      return !Objects.equals(existingTotal, recalculatedTotal);
```

Comment:
This expression does not cover the case when the parts (`adjustmentTotal`, `subTotal`) have changed, but the `total` remained the same. In this case, the invoice line will not be updated and the value of `adjustmentTotal` for `invoiceLines` retrieved via `GET by Query` will be different from what is in storage.